### PR TITLE
Add a way to use factories to create Cocina objects

### PR DESCRIPTION
**Pull request by jcoyne**
## Why was this change made? 🤔

To reduce the amount of boilerplate in tests.

## How was this change tested? 🤨

⚡ ⚠ If this change has cross service impact (including writing to shared file systems or interacting with other SDR APIs), ***run [integration tests](https://github.com/sul-dlss/infrastructure-integration-test)*** and/or test in [stage|qa] environment, in addition to specs. ⚡

⚡ ⚠ If this change updates the Argo UI, run all integration tests that use Argo and create a PR on the integration test repo to fix anything this change breaks. ⚡

### CODE REVIEW

File: spec/factories/apos.rb [hunk 1]
```diff
@@ -1,7 +1,7 @@
 # frozen_string_literal: true
 
 FactoryBot.define do
-  factory :apo, class: 'Cocina::Models::RequestAdminPolicy' do
+  factory :persisted_apo, class: 'Cocina::Models::RequestAdminPolicy' do
```
Comment (justinlittman) [inline spec/factories/apos.rb:4]: Why is this `persisted_apo`? The name choice doesn't make sense to me.

File: spec/factories/collections.rb [hunk 2]
```diff
@@ -25,4 +25,34 @@
...
+      )
+    end
+
+    admin_policy_id { 'druid:hv992ry2431' }
+    external_identifier { 'druid:bc123df4568' }
+
+    label { 'test object' }
+    type { Cocina::Models::ObjectType.collection }
+    title { 'my collection' }
+    purl { 'https://purl.stanford.edu/bc123df4568' }
```
Comment (justinlittman) [inline spec/factories/collections.rb:56]: Can the purl be derived from the `external_identifier`?

File: spec/factories/items.rb [hunk 3]
```diff
@@ -1,7 +1,7 @@
 # frozen_string_literal: true
 
 FactoryBot.define do
-  factory :item, class: 'Cocina::Models::RequestDRO' do
+  factory :persisted_item, class: 'Cocina::Models::RequestDRO' do
```
Comment (justinlittman) [inline spec/factories/items.rb:4]: For naming consistency / clarity, can we use `dro` instead of `item`?

File: spec/factories/apos.rb (hunk 1 above)
Comment (justinlittman) [reply spec/factories/apos.rb:4]: Oh wait, this is actually saved to the datastore. The "request" threw me. Can you please add comments to make the distinction between persisted and not persisted clear?

File: spec/components/show/item/overview_component_spec.rb [hunk 4]
```diff
@@ -7,20 +7,7 @@
...
-                              title: [{ value: 'my dro' }],
-                              purl: 'https://purl.stanford.edu/bc234fg5678'
-                            },
-                            access: {},
-                            identification: {},
-                            structural: {},
-                            administrative: {
-                              hasAdminPolicy: 'druid:hv992ry2431'
-                            })
+    build_for_repository(:item)
```
Comment (justinlittman) [inline spec/components/show/item/overview_component_spec.rb:10]: What is the distinction between `build_for_repository` and `create_for_repository`? Also, the "_for_repository" feels a bit odd. "_cocina" instead?

File: spec/factories/items.rb (hunk 3 above)
Comment (justinlittman) [reply spec/factories/items.rb:4]: Oh, I take that back. This does following naming conventions https://github.com/sul-dlss/cocina-models#usage-conventions

File: spec/support/build_strategy_repository_pattern.rb [hunk 5]
```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+class BuildStategyForRepositoryPattern
```
Comment (justinlittman) [inline spec/support/build_strategy_repository_pattern.rb:3]: Typo in class name.

Class documentation?

**Review by justinlittman — COMMENTED** [review]
Mostly naming suggestions / requests. Will it make sense to add to cocina models so that can be easily used in other repos? Or just copy to other repos?

**Review by jcoyne — COMMENTED** [review]
The create_ puts it in DSA, the build_ just builds one in memory

**Comment by justinlittman** [comment]
How does that create_/build_ relate to persisted_?

**Comment by jcoyne** [comment]
@justinlittman `create` and `build` are the default functions from FactoryBot.  So this keeps that convention for naming.

File: spec/components/show/item/overview_component_spec.rb (hunk 4 above)
Comment (jcoyne) [reply spec/components/show/item/overview_component_spec.rb:10]: We've had `create_for_repository` in this repo for 2 years. I think it's pretty good. It was copied from Hyrax: https://github.com/samvera/hyrax/blob/573f1ce2f9eea19b32bf9905b1725db63d5be730/spec/support/create_for_repository.rb

**Pull request closed without merging**